src/modules/InstURL.rb

```diff
@@ -142,232 +83,45 @@ def RewriteCDUrl(url)
-          options = Ops.add(
-            Ops.add("device=/dev/", partition),
-            "&filesystem=auto"
-          )
-        else
-          Builtins.y2error("no partition specified")
-        end
+      if parts.empty?
+        # Make it compatible with the current behaviour when
+        # install.inf does not exist.
+        log.warn "No URL specified through ZyppRepoURL"
+        parts = URL.Parse("cd:///")
       end
 
-      Ops.set(url_tokens, "scheme", instmode)
+      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
 
-      if @is_network
-        username = Linuxrc.InstallInf("Username")
-        if username != nil && username != ""
-          Ops.set(url_tokens, "user", username)
-          password = Linuxrc.InstallInf("Password")
-          if password != nil && password != ""
-            Ops.set(url_tokens, "pass", password)
-          end
-        end
-        servername = Linuxrc.InstallInf("ServerName")
-        server = Linuxrc.InstallInf("Server")
-        serverip = Linuxrc.InstallInf("ServerIP")
-
-        if servername != nil && servername != ""
-          Ops.set(url_tokens, "host", servername)
-        elsif server != nil && server != ""
-          Ops.set(url_tokens, "host", server)
-        elsif serverip != nil && serverip != ""
-          Ops.set(url_tokens, "host", serverip)
-        end
-      end # is_network
-
-      isoimg = ""
-      serverdir = Linuxrc.InstallInf("Serverdir")
-      if Linuxrc.InstallInf("SourceType") == "file"
-        if serverdir != "" && serverdir != nil
-          sd_items = Builtins.splitstring(serverdir, "/")
-          sd_items = Builtins.filter(sd_items) { |i| i != "" }
-          last = Ops.subtract(Builtins.size(sd_items), 1)
-          isoimg = Ops.get(sd_items, last, "")
-          Ops.set(sd_items, last, "")
-          serverdir = Builtins.mergestring(sd_items, "/")
-        end
-      end
-
-      # 	if (((instmode == "hd") || is_network)				// if serverdir needed
-      # 	    && ((serverdir != nil) && (serverdir != "")))		// and is valid
-      # 	{
-      # 	    // for smb mounts it is usual to not have a leading slash
-      # 	    if (substring (serverdir, 0, 1) != "/")
-      # 		serverdir = "/" + serverdir;
-      # 	}
-      share = Linuxrc.InstallInf("Share")
-
-      if extra_dir != ""
-        if serverdir != nil
-          # avoid too many slashes
-          if Builtins.findlastof(serverdir, "/") ==
-              Ops.subtract(Builtins.size(serverdir), 1)
-            serverdir = Builtins.substring(
-              serverdir,
-              0,
-              Ops.subtract(Builtins.size(serverdir), 1)
-            )
-          end
-
-          slash = ""
-          slash = "/" if Builtins.substring(extra_dir, 0, 1) != "/"
-          serverdir = Ops.add(Ops.add(serverdir, slash), extra_dir)
-          slash = ""
-        else
-          serverdir = extra_dir
-        end
+      if parts["scheme"] == "https" && !SSLVerificationEnabled()
```

Comment:
`SSLVerificationEnabled` checks whether `Linuxrc.InstallInf("ssl_verify") != "no"` but ssl_verify is not documented in https://en.opensuse.org/SDB:Linuxrc_install.inf so there are potential bugs when checking/using "0" or "false".